modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcMultiStatementSelfTest.java

```diff
@@ -52,48 +56,321 @@ public void setupTables() throws Exception {
+
+        res = stmt.execute("SELECT 1; SELECT 1;");
+        assertEquals(true, res);
+        assertEquals(2, getResultSetSize());
+    }
+
+    /** Check update count invariants. */
+    @Test
+    public void testUpdCountMisc() throws Exception {
+        // pure select case
+        stmt.execute("SELECT 1; SELECT 1;");
+        assertEquals(-1, stmt.getUpdateCount());
+
+        ResultSet rs = stmt.getResultSet();
+        assertEquals(-1, stmt.getUpdateCount());
+
+        rs.next();
+        assertEquals(-1, stmt.getUpdateCount());
+
+        stmt.getMoreResults();
+        assertTrue(rs.isClosed());
+        rs = stmt.getResultSet();
+        assertEquals(-1, stmt.getUpdateCount());
+
+        rs.next();
+        assertEquals(-1, stmt.getUpdateCount());
+
+        // empty result
+        stmt.execute("SELECT ID FROM TEST_TX WHERE ID=1000;");
+        assertEquals(-1, stmt.getUpdateCount());
+    }
+
+    @Test
+    public void testUpdCountNoMoreResults() throws Exception {
+        stmt.execute("INSERT INTO TEST_TX VALUES (5, 5, '5'), (6, 5, '5');");
+        assertEquals(2, stmt.getUpdateCount());
+        stmt.getMoreResults();
+        assertEquals(-1, stmt.getUpdateCount());
+        stmt.getResultSet();
+        assertEquals(-1, stmt.getUpdateCount());
+    }
+
+    @Test
+    public void testMixedQueriesUpdCount() throws Exception {
+        stmt.execute("SELECT 1; INSERT INTO TEST_TX VALUES (7, 5, '5');");
+        stmt.getMoreResults();
+        assertEquals(1, stmt.getUpdateCount());
+        assertEquals(1, stmt.getUpdateCount());
+        stmt.getMoreResults();
+        assertEquals(-1, stmt.getUpdateCount());
+
+        stmt.execute("DROP TABLE IF EXISTS TEST_TX; DROP TABLE IF EXISTS PUBLIC.TRANSACTIONS;");
+        assertEquals(0, stmt.getUpdateCount());
+
+        stmt.execute("CREATE TABLE TEST_TX (ID INT PRIMARY KEY, AGE INT, NAME VARCHAR) ");
+        assertEquals(0, stmt.getUpdateCount());
+    }
+
+    @Test
+    public void testUpdCountAfterError() throws Exception {
+        try {
+            stmt.execute("INSERT INTO NOT_EXIST VALUES (3, 17, 'James');");
+        } catch (Throwable ignored) {
+            // No op.
+        }
+        ResultSet rs = stmt.getResultSet();
+        assertNull(rs);
+        assertEquals(-1, stmt.getUpdateCount());
+    }
+
+    @Test
+    public void testResultsFromExecuteBatch() throws Exception {
+        stmt.addBatch("INSERT INTO TEST_TX VALUES (7, 25, 'Michel');");
+        stmt.addBatch("INSERT INTO TEST_TX VALUES (8, 25, 'Michel');");
+        int[] arr = stmt.executeBatch();
+
+        assertEquals(2, arr.length);
+        assertArrayEquals(new int[]{1, 1}, arr);
+        assertEquals(-1, stmt.getUpdateCount());
+        assertFalse(stmt.getMoreResults());
     }
 
     /**
      * Sanity test for scripts, containing empty statements are handled correctly.
      */
     @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-21081")
```

Comment:
I think we can enable this test, we have already discussed that there is only one problematic case here ";" (empty statement), we have test for this IgniteSqlParaserTest#testEmptyStatements.
Other cases should work fine.

Reply:
what about : ";;;SELECT 1 + 2", seems it valid statement ?

Reply:
I've updated description in the linked ticket.